linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -379,6 +379,23 @@ final class Emitter(config: Emitter.Config) {
 
     // Class definition
     if (linkedClass.hasInstances && kind.isAnyNonNativeClass) {
+      /* Is this class compiled as an ECMAScript `class`?
+       *
+       * See JSGen.useClassesForRegularClasses for the rationale here.
+       * Accessing `ancestors` without cache invalidation is fine because it
+       * is part of the identity of a class for its cache in the first place.
+       *
+       * Note that `useClassesForRegularClasses` implies
+       * `useClassesForJSClassesAndThrowables`, so the short-cut is valid
+       */
+      val useESClass = if (jsGen.useClassesForRegularClasses) {
+        assert(jsGen.useClassesForJSClassesAndThrowables)
+        true
+      } else {
+        jsGen.useClassesForJSClassesAndThrowables &&
+        (kind.isJSClass || linkedClass.ancestors.contains(ThrowableClass))
```

Comment:
This check here is not consistent with `shouldExtendJSError`: it checks also if the super class of `Throwable` is `jl.Object`. In practice, this is probably unimportant. However, a comment referring to `shouldExtendJSError` and saying that this is a correct approximation (in the worst case we just don't optimize some classes) might be a good idea.